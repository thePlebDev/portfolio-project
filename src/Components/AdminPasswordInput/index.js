import React from 'react';

import styled from 'styled-components';
import PropTypes from 'prop-types'

const Container = styled.div`
  margin-top:20px;
  margin-bottom:10px;
`
const Input = styled.input`
  width:60%;
  padding:10px;
  border:${props=>props.state ? '1px solid red':'1px solid black'};
`


const AdminPasswordInput = ({state,handleChange,error})=>{


  return(
    <Container data-testid="container">
      <label id="password" >
        <Input htmlFor="password" data-testid="password-input" state={error} name="password" type="password" value={state} onChange={(e)=>handleChange(e)} placeholder='password'/>
      </label>
    </Container>
  )
}

AdminPasswordInput.propTypes ={
  state:PropTypes.string,
  handleChange:PropTypes.func,
  error:PropTypes.string

}

export default AdminPasswordInput
