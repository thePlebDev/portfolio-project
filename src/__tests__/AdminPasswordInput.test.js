import React from 'react';
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme';
import AdminPasswordInput from '../Components/AdminPasswordInput'

// import { configure } from 'enzyme';
// import Adapter from 'enzyme-adapter-react-16';
//
// configure({ adapter: new Adapter() });


//1) (DONE) test snapshow
//2) testing props, what goes in and what comes out
//3) testing the type of the props
//4) event testing
//5) testing conditions, all posible conditions that can happen to the component
//6) state testing,1) check state before event. 2)check state after event.


describe('testing the AdminPasswordInput component',()=>{
  it('should test the snapshot properly',()=>{
    const tree = renderer.create(<AdminPasswordInput/>).toJSON();
    expect(tree).toMatchSnapshot()
  })
  it('should render properly',()=>{
    const wrapper = shallow(<AdminPasswordInput/>)
    const container = wrapper.find('[data-testid="container"]')
    const input = wrapper.find('[data-testid="input"]')
    expect(input.length).toBe(1)
    expect(container.length).toBe(1)
  })
  it('should render with the proper values',()=>{
    let state = 'this'
    const mockHandleChange =jest.fn(x=>42 +x)

    const wrapper = shallow(<AdminPasswordInput state={state} handleChange={mockHandleChange} error={'there do be errors in here'}/>)
    let input = wrapper.find('[data-testid="input"]')
    input.simulate('change',{
      target:{name:'password',value:'Jack'},
    })
    expect(input.props().state).toBe('there do be errors in here')
    expect(mockHandleChange.mock.calls.length).toBe(1)
  })
})
