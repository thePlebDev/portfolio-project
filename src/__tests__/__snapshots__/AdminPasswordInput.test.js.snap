// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the AdminPasswordInput component should test the snapshot properly 1`] = `
<div
  className="sc-bdnylx clAjGN"
  data-testid="container"
>
  <label
    id="password"
  >
    <input
      className="sc-gtssRu ehmHiR"
      data-testid="input"
      for="password"
      name="password"
      onChange={[Function]}
      placeholder="password"
      type="password"
    />
  </label>
</div>
`;
